--- upstream/stellaParser.g4
+++ stella_checker/syntax/grammar/stellaParser.g4
@@ -42,7 +42,7 @@
     | 'true'                             # ConstTrue
     | 'false'                            # ConstFalse
     | 'unit'                             # ConstUnit
-    | n = INTEGER                        # ConstInt
+    | sign = '-'? n = INTEGER             # ConstInt
     | mem = MemoryAddress                # ConstMemory
     | name = StellaIdent                 # Var
     // expr
@@ -88,7 +88,7 @@
     | 'match' expr '{' (
         cases += matchCase ('|' cases += matchCase)*
     )? '}'                                         # Match
-    | '[' (exprs += expr (',' exprs += expr))? ']' # List
+    | '[' (exprs += expr (',' exprs += expr)*)? ']' # List
     // expr
     | left = expr '<' right = expr  # LessThan
     | left = expr '<=' right = expr # LessThanOrEqual
@@ -154,7 +154,7 @@
             ',' fieldTypes += variantFieldType
         )*
     )? '|>'                                                     # TypeVariant
-    | '[' (types += stellatype (',' types += stellatype)*)? ']' # TypeList
+    | '[' types += stellatype ']' # TypeList
     | 'Unit'                                                    # TypeUnit
     | 'Top'                                                     # TypeTop
     | '&' type_=stellatype                                      # TypeRef
--- upstream/stellaLexer.g4
+++ stella_checker/syntax/grammar/stellaLexer.g4
@@ -99,1 +99,1 @@
-MemoryAddress: '<0x' (DIGIT|[a-f])+ '>';
+MemoryAddress: '<0x' (DIGIT|[a-fA-F])+ '>';
